File: src/main/java/com/clinica/atendimento/service/messaging/producer/ResumirProducer.java
package com.clinica.atendimento.service.messaging.producer;

import org.apache.kafka.common.serialization.StringSerializer;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

@Component
public class ResumirProducer extends AbstractProducer<String, String> {

    public ResumirProducer(@Value("${kafka.url}") String bootstrapServer,
                           @Value("${kafka.topico.resumir}") String topico) {
        super(bootstrapServer, topico, StringSerializer.class, StringSerializer.class);
    }
}

File: src/main/kotlin/com/clinica/atendimento/service/producer/ResumirProducer.kt
package com.clinica.atendimento.service.producer

import org.apache.kafka.common.serialization.StringSerializer
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component

@Component
class ResumirProducer(
    @Value("\${kafka.url}") bootstrapServer: String,
    @Value("\${kafka.topico.resumir}") topico: String
) : AbstractProducer<String, String>(
    bootstrapServer,
    topico,
    StringSerializer::class.java,
    StringSerializer::class.java
)